sbe-tool/src/main/java/uk/co/real_logic/sbe/generation/java/JavaGenerator.java

```diff
@@ -1858,6 +1905,32 @@ private void generateCharArrayEncodeMethods(
                 propertyName,
                 fieldLength,
                 offset));
+            sb.append(String.format("\n" +
+                indent + "    public %1$s %2$s(final CharSequence src)\n" +
+                indent + "    {\n" +
+                indent + "        final int length = %3$d;\n" +
+                indent + "        final int srcLength = src.length();\n" +
+                indent + "        if (srcLength > length)\n" +
+                indent + "        {\n" +
+                indent + "            throw new IndexOutOfBoundsException(" +
+                "\"CharSequence too large for copy: byte length=\" + srcLength);\n" +
+                indent + "        }\n\n" +
+                indent + "        for (int i = 0; i < srcLength; ++i)\n" +
+                indent + "        {\n" +
+                indent + "            final char charValue = src.charAt(i);\n" +
+                indent + "            final byte byteValue = charValue > 127 ? (byte)'?' : (byte)charValue;\n" +
```

Comment:
Is this the right thing to do with all non-ASCII encodings? Wouldn't it be better to simply copy the char value without the check?